src/main/java/org/opensearch/security/resources/ResourceActionGroupsHelper.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.security.resources;
+
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.opensearch.security.securityconf.FlattenedActionGroups;
+import org.opensearch.security.securityconf.impl.CType;
+import org.opensearch.security.securityconf.impl.SecurityDynamicConfiguration;
+import org.opensearch.security.securityconf.impl.v7.ActionGroupsV7;
+
+import org.yaml.snakeyaml.Yaml;
+
+// CS-SUPPRESS-SINGLE: RegexpSingleline get Resource Sharing Extensions
```

Comment:
We should probably remove the check to keep out the word extension soon. Given that the extensions effort is not under active development I don't think it makes sense to keep it, especially as we make modifications for extensible/extending plugin.